src/actions/ProvidersActions.ts

```diff
@@ -0,0 +1,45 @@
+import { Notify } from 'quasar';
+import { getListOfProvidersQuery } from '../services';
+import {
+  useQuery,
+  actionCallbackReturnTypes,
+  actionCallbackParamsTypes,
+} from '../utils/apollo';
+
+export const getListOfProviders = ({
+  limit = 10,
+  offset = 0,
+  query = '',
+}: actionCallbackParamsTypes) => {
+  const name = `%${query}%`;
+
+  return new Promise<actionCallbackReturnTypes>((resolve, reject) => {
+    const variables = {
+      name,
+      offset,
+      limit,
+    };
+
+    useQuery(getListOfProvidersQuery, variables)
+      .then(({ providers, providers_aggregate }) => {
+        if (!providers || !providers[0]) {
+          Notify.create({
+            message: 'No se encontraron resultados para la busqueda actual',
+            type: 'negative',
+          });
+
+          reject(null);
+          return null;
```

Comment:
Hmm isn't this return unreacheable? Like the scope will finish on the resolve line, please correct me if I am wrong

Reply:
I've thought that too, but without the return the code continue and then I receive the error notification and the success notification

Reply:
OK, if you tried I am fine, thanks for confirm!